File: DeckOfCards-ObjC/DeckOfCards/ViewController.m
//
//  ViewController.m
//  DeckOfCards
//
//  Created by Andrea Prearo on 7/15/16.
//
//

#import "ViewController.h"
#import "Deck.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    Deck *deck = [Deck create];
    NSLog(@"New deck ready.");
    [deck shuffle];
    NSLog(@"Shuffled deck.");
    NSUInteger size = deck.size;
    for (int i = 0; i < size; i++) {
        NSLog(@"Dealt: %@. %lu card(s) dealt. %d remaining.", [deck dealOne],
              i+1, deck.size);
    }
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

@end

File: DeckOfCards-Swift/DeckOfCards/ViewController.swift
//
//  ViewController.swift
//  DeckOfCards
//
//  Created by Andrea Prearo on 12/30/14.
//  Copyright (c) 2014 Andrea Prearo. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        let deck = Deck()
        print("New deck ready.")
        deck.shuffle()
        print("Shuffled deck.")
        let size = deck.size
        for i in 0..<size {
            print("Dealt: \(deck.dealOne()!.description). \(i+1) card(s) dealt. \(deck.size) remaining.")
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

}

